### PR TITLE
Restore condition with non-generic type parameter

### LINKED ISSUE #2168: SRTP `or` condition disappear when non-generic type is used
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMBLAdugYmEgM6IBOuAhkQF6IA8YAejWAO4AWi%2BYACjYdYlMADFKsALYApMrHwARRADMaAV3RYyASjBoBZLDSy5oYaYmkAjKhKlyF-NMLABaAHxh5ilu09dbwArZzAAXgAdSkj%2BCDBQxTAAH29ffAA6AAcaSgoUtOcM8ipaBmZhT0gAGnAoCiwzfABzMlQwAG06iFB4%2BMgAEiwATyzkNEgCLBru-oGANzoNcc7ZvoAGWr6%2ByABJfCR8LABlXEYZ7fiAFjWwAF1ZgF8t%2BN7twZGx9smji-eFpYrLqXCAARhel0gAFkaAAPAAyxHhfGaWG4f0uoIATOtbg8%2Bs9Zm8dkNRitIHx4AB5VSI-CIY4jTAYuaLdDLdrAkFYiHvACiBxpdOQvJ26FUkDxT15xLmn3JNlgsHQLIggzZHLQXMuAGZRf19hRKFhxAQ6AA5RCcIj01XxLCUZZSgky24fMnfRXKu3qwGc24QK76tXHazmZWKACCWAF1NpxDtEHU6AozvihL6srVpK%2BEy9Kv1vvZQIDYAArMGoMcctBEAAhNSiRAABVyNCsWCoibADqdl3x6ddkJzCqVBbdAOL-pBYAAbJXINWaLWG6om-DYJwqNAaBR9vNYDuzIpu73EGmIBnXhP5Z6xz7J5rVjOAOwLpcrxuURAAVSyY0oHc93wA8j1wE9K2TVN%2B2lIkbw9PN70LR8SxnAAOd8a3rL9EAAYXQXcyFwxRjEdaAsFEbsoPPGCXTg4db0Q71kI1VCQQATkw5dsLXb8oWsOxKCouhoO2AdLyHf5GKgfMH1Y6cQVBTZS0XLDVybYj0Ag0tqIvMArx6eDcxkpCJ3krVS1BcEVI-RBI1UTtKGI6RpBoU9HRosTYMzIzR2Ysy-QsmdsS4z9eIZMM4C0-BhJTTy%2BnE-TJJJaTIFkljAufRS9RsrD7Mc0NpHDaL3L7Ly6J8hiEJM-zh3MrLMSDFT9kOLAqXwAAVShhgAdVwNFYtEhLvOvKrjLS0y6sy7VtlBCtcu4yMpA0A4VCIIrHNK%2BL4kSgywCzKAR2%2BKY5Omyz5xUmFYV2VQOt4fA%2BTi45uFELA%2BvgAbKyuXFaMHeipOqn5pgyqcgsUt9LrhfZVFwWEqQA0xRD5WEsm-MgyHAmLKzLH7yr%2ByqAfGk6QafGa%2BlBDDIdhAAlRA4EoeB3s%2B0tvr0vaDvdInfhJtjMU4qnafp%2BBzQ0WwqBpXZO2kNpK1BNnkrlQHpC0MwbUQcRRFcxoqA6j0eYUy4cQXQXRHgKEVdwNWNcoLXNoXaBuDbciqAAfU4XAPvReX-pSwHiYC0GGu2LFrJnaE4SWygaGGKlKERYwmfRL7ceGirRsJ8l-amwOyfiLEeSpyPo9j%2BOsBFsXKAlqWZcs72Cd98ble0S3iGt22db1gPSdLLEcrDouY7j3BjHN5urc10w7ZUh2ncct2PYGuv04bzPua73ng6asOroANUBOsCHgAhmkT7tWd%2BiSfcVrngfXg3g-m7e4XEFbyMxg%2BDmP0-k6XwyxtX2%2B2du4zixBdJ%2BsIlCwCwAAcWwMjVGiB0aY2-qWHGv99q%2BTvLVf49Vc4QCxBDbeFs1Z1i0tAAA1nWKOFDsBkHascdseFlSiyxjpES20IC7QVtmVK6U75g0NpTMOlprTEAbFgLcfBdZjBUDDfA-VMaRgOPxCuNcZy6QvklK%2BPDAZ8KARvPoWJ%2BZhwANKIEQFkG6d0%2BD7AYVYYUg0OH3BGn-DOx0176Pvn0HUylwGPSKmQbgKCZznzxpfeu18-Ljk8QI7YOpQ4gkXMfTAS1miiz4G9RANgoQIBFGwuK6COZHSYtEnBZ0Zw6gLv3IgzR8Av3wG-RQpxammA0N%2BDqsAWoZNMJjRxhTMElNOjnUsOo%2B6JMjNoKg%2BAenzEQBuFoKjBKyIIAokifTNHswGTVUpJJcEjK3ok0eZhhTwnbDYeANB8KwAxi0ERat1lhK0REnR409FlOGRUx%2BiSlDDxoDYTA-jh7cGOMMI4cIHmp3xsvSJWCdmsnKSCHUYDElmIsV0o4%2BwqE0AaUnfJQ0douIwf-WFQzgGIsIYk0h2LyHCjIEtWAK1haIM7GbYhxB4FowxmsysZ5%2BnEsGfrWJ3ihGUtyOpb8PyyDQGoEVaZLKfzyMUCoaABEo7HlYeo9hfK3ECv4UHbxxjEnUxoC0qQ5CIUErTq4leJLBX6uuL4xJjJqDkRyUgC1nCnh1DuC8SYZBxAY3aLpR4QA)

#### Code

```fsharp
let inline deserialize< ^a when ( ^a or FromJsonDefaults) : (static member FromJson :  ^a -> Json< ^a>)> json =
    json |> Json.parse |> Json.deserialize< ^a>
```
            

#### Result

```fsharp
let inline deserialize< ^a when ^a: (static member FromJson: ^a -> Json< ^a >)> json =
    json |> Json.parse |> Json.deserialize< ^a>

```
            
#### Problem description

See title.

Note, that `let inline deserialize< ^a when ( ^a or ^b) : (static member FromJson :  ^a -> Json< ^a>)> json =
    json |> Json.parse |> Json.deserialize< ^a>` works correctly.

Related to #984.

#### Extra information

- [x] The formatted result breaks by code.
- [ ] The formatted result gives compiler warnings.
- [x] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 2022-03-27T13:24:16Z - fbd4b7ff24260788588325d3fe9cd7d62d9aadf0

```fsharp
    { config with
                MaxValueBindingWidth = 40 }
```

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

### PR DESCRIPTION
**Pull request by jindraivanek**

### CODE REVIEW

**Comment by jindraivanek** [comment]
Should be working solution, but probably need some DRY treatment.

**Comment by jindraivanek** [comment]
Fix for #2168

**Comment by jindraivanek** [comment]
ready for review

File: src/Fantomas/Context.fs [hunk 1]
```diff
@@ -599,6 +599,21 @@ let internal colPreEx f2 f1 (c: seq<'T>) f (ctx: Context) =
     else
         colEx f1 c f (f2 ctx)
 
+/// Similar to col, but apply two more functions fStart, fEnd at the beginning and the end if the input sequence is bigger thn one item
+let internal colSurr fStart fEnd f1 (c: list<'T>) f (ctx: Context) =
```
Comment (nojaf) [inline src/Fantomas/Context.fs:603]: I was a bit unsure about this abstraction at first.
I'm not entirely convinced `col f1 c f` should be part of this abstraction.
At the same time, I have no example code where a more general abstraction would make sense, so let's just go with this.

File: src/Fantomas/Context.fs [hunk 2]
```diff
@@ -599,6 +599,21 @@ let internal colPreEx f2 f1 (c: seq<'T>) f (ctx: Context) =
...
+        ctx
+    else
+        (col f1 c f
+         |> fun g ->
+             if (List.moreThanOne c) then
+                 fStart +> g +> fEnd
+             else
+                 g)
+            ctx
+
```
Comment (nojaf) [inline src/Fantomas/Context.fs:614]: I'll remove these additional newlines, triggers me a bit 🙈.

**Review by nojaf — APPROVED** [review]
Solid as usual! Thanks buddy!

**Comment by nojaf** [comment]
I changed the title due to this guidance, https://github.com/fsprojects/fantomas/blob/master/CONTRIBUTING.md#pull-request-title
Nitpick, I know 😅

**Pull request merged**